src/Pact/Types/Type.hs

```diff
@@ -297,9 +299,9 @@ canUnifyWith a b = a == b
 -- valid in a, etc. [] in a subset is a wildcard.
 canUnifyPartial :: SchemaPartial -> SchemaPartial -> Bool
 canUnifyPartial SPFull SPFull = True
-canUnifyPartial SPPartial SPPartial = True
-canUnifyPartial SPFull SPPartial = True -- full can always satisfy subset
-canUnifyPartial SPPartial SPFull = False -- partial cannot satisfy full
+canUnifyPartial (SPPartial a) (SPPartial b) = null b || all (`elem` a) b
```

Comment:
Note: if we used sets instead of lists the cost here would be `O(n + m)` instead of `O(n * m)`

Reply:
Also, `subset` is easier to read